docs/source/config/ui.rst

```diff
@@ -0,0 +1,272 @@
+ui.yaml
+=======
+
+This configuration file allows customizing the UI to match the local beamline.
+It allows configuring widgets for controlling gonio motors, on-axis sample viewer, hutch video streams and more.
+The ``ui.yaml`` contains following sections:
+
+* sample_view_
+* sample_view_video_controls_
+* beamline_setup_
+* camera_setup_
+
+Each section groups settings for related UI features.
+See below an `example  ui.yaml <ui_yaml_example_>`_ file.
+
+sample_view
+-----------
+
+This section configures widgets for controlling gonio motors used for sample alignment.
+The section has the following syntax:
+
+.. code-block:: yaml
+
+    sample_view:
+      id: sample_view
+      components:
+        - label: <motor_label>
+          attribute: <beamline_attribute>
+          role: <motor_role>
+          step: <step_size>
+          precision: <value_precision>
+          suffix: <value_suffix>
+
+
+The ``components`` key contains a list of gonio motor specifications.
+Each motor specification have the following keys:
+
++-----------+-------------------------------------+
+| key       | purpose                             |
++===========+=====================================+
+| attribute | motor's beamline attribute id       |
++-----------+-------------------------------------+
+| role      | motor's sample movement role        |
++-----------+-------------------------------------+
+| label     | widget's text label                 |
++-----------+-------------------------------------+
+| step      | widget's single step size           |
++-----------+-------------------------------------+
+| precision | widget's position display precision |
++-----------+-------------------------------------+
+| suffix    | widget's position units suffix      |
++-----------+-------------------------------------+
+
+The ``attribute`` key specifies the underlying motor's adapter object, via it's beamline attribute ID.
+The ``role`` key specifies the physical move the motor performs on the sample.
+For example, motor with the role *omega* rotates sample around the omega axis.
+
+Valid roles are:
```

Comment:
Roles can be anything its not actually finite list, perhaps Common roles are:

Reply:
Aha, changed to "Common roles".